sql/binding_contexts.cc

```diff
@@ -169,13 +169,24 @@ doGetFunction(const Utf8String & tableName,
 
     auto fn = outer.doGetColumnFunction(functionName);
 
-    return {[=] (const std::vector<ExpressionValue> & args,
+    if (fn)
+    {
+         return {[=] (const std::vector<ExpressionValue> & args,
                  const SqlRowScope & context)
             {
                 auto & col = static_cast<const ColumnContext &>(context);
                 return fn(col.columnName, args);
             },
             std::make_shared<Utf8StringValueInfo>()};
+    }
+
+    auto sqlfn = SqlBindingScope::doGetFunction(tableName, functionName, args);
```

Comment:
After discussion with Mathieu, some check will have to be written to ensure that the training does not fail badly on column expression with functions.  I will do that as part of MLDB-1104.